tests/upgrades/test_host.py

```diff
@@ -16,107 +16,185 @@
+        if not host_name:
+            return googleclient.get_vm(name='{}'.format(self.fullhostname.replace('.', '-')))
+        else:
+            return googleclient.get_vm(name='{}'.format(host_name.replace('.', '-')))
 
     @pytest.mark.pre_upgrade
+    @pytest.mark.tier1
     def test_pre_create_gce_cr_and_host(
-        self, arch_os_domain, function_org, gce_cert, save_test_data
+        self,
+        class_host,
+        module_puppet_org,
+        module_puppet_loc,
+        module_gce_compute_puppet,
+        module_puppet_domain,
+        gce_hostgroup_puppet,
+        module_gce_finishimg_puppet,
+        googleclient,
+        save_test_data,
     ):
-        """Create GCE Compute Resource
+        """Host can be provisioned on Google Cloud
+
+        :id: 889975f2-56ca-4584-95a7-21c513969630
 
-        :id: preupgrade-ef82143d-efef-49b2-9702-93d67ef6804c
+        :CaseLevel: Component
 
-        :steps: In Preupgrade Satellite, create GCE Compute Resource
+        ::CaseImportance: Critical
+
+        :steps:
+            1. Create a GCE Compute Resource
+            2. Create a Host-group with all the Global and Foreman entities but
+                without Compute Profile, required to provision a host
+            3. Provision a Host on Google Cloud using above GCE CR and Host-group
 
-        :expectedresults: The GCE CR created successfully
+        :expectedresults:
+            1. The host should be provisioned on Google Compute Engine
+            2. The host name should be the same as given in data to provision the host
+            3. The host should show Installed status for provisioned host
         """
-        arch, os, domain_name = arch_os_domain
-        cr_name = gen_string('alpha')
-        loc = entities.Location().create()
-        json_key = json.dumps(gce_cert, indent=2)
-        with Session('gce_upgrade_tests') as session:
-            # Compute Resource Create and Assertions
-            session.computeresource.create(
-                {
-                    'name': cr_name,
-                    'provider': FOREMAN_PROVIDERS['google'],
-                    'provider_content.json_key': json_key,
-                    'provider_content.zone.value': settings.gce.zone,
-                    'organizations.resources.assigned': [function_org.name],
-                    'locations.resources.assigned': [loc.name],
-                }
-            )
-        gce_cr = entities.AbstractComputeResource().search(query={'search': f'name={cr_name}'})[0]
-        gce_img = entities.Image(
-            architecture=arch,
-            compute_resource=gce_cr,
-            name='autoupgrade_gce_img',
-            operatingsystem=os,
-            username='gceautou',
-            uuid=LATEST_RHEL7_GCE_IMG_UUID,
-        ).create()
+        assert class_host.name == self.fullhostname
+        assert class_host.build_status_label == 'Installed'
+        assert class_host.ip == self.google_host(googleclient, None)
```

Comment:
Need not to be explicit  with `None` as its none by default in function defination.